src/main/kotlin/blackjack/domain/GameResult.kt

```diff
@@ -1,18 +1,23 @@
 package blackjack.domain
 
+import blackjack.domain.player.Player
 import blackjack.domain.player.User
 
 class GameResult(players: List<User>, dealer: User) {
-    val players: Map<User, ResultType> = initPlayerResult(players, dealer)
-    val dealer: Map<ResultType, Int>
-        get() {
-            return mapOf(
-                Pair(ResultType.WIN, players.count { it.value == ResultType.LOSE }),
-                Pair(ResultType.DRAW, players.count { it.value == ResultType.DRAW }),
-                Pair(ResultType.LOSE, players.count { it.value == ResultType.WIN }),
-            )
-        }
+    val players: Map<User, BettingMoney> = initPlayerResult(players, dealer)
+    val dealer: BettingMoney = initDealerResult()
+
+    private fun initDealerResult() = BettingMoneys(players.values.toList()).sum()
 
     private fun initPlayerResult(players: List<User>, dealer: User) =
-        players.map { Pair(it, it.calculateScore().compareTo(dealer.calculateScore())) }.toMap()
+        players.map { Pair(it, calculateProfit(it, dealer)) }.toMap()
+
+    private fun calculateProfit(_player: User, dealer: User): BettingMoney {
```

Comment:
제가 잘못 알고 있을 수도 있는데..
`Backing Properties`를 사용하는 경우 이외에 `_`로 시작하는 명명 규칙은 따로 없었던 것 같아요.

```suggestion
    private fun calculateProfit(player: User, dealer: User): BettingMoney {
```

파라미터로 넘어간 값과 지역 변수 이름이 같으면 `shadowing` 때문에 warning이 발생하긴 하겠네요..
이 경우 어떻게 변수명을 지정하면 좋을까요? 🤔 

그냥 _로 파라미터를 넘겨줘도 warning이 뜨지 않으니 사용하는게 좋을까요?

Reply:
파라미터로도 플레이어란것을 명시해줘야하고, 함수 스코프 안에서도 플레이어라고 표현하는 것이 적절하다고 생각해서 사용했습니다.
함수 스코프 안에서는 gamePlayer라고 지정하려다가 오히려 플레이어와 게임플레이어가 다른가?라는 혼란을 줄 수 있을것 같더라구요.
이 부분에 대한 컨벤션은 저도 잘 모르겠지만 사용해봤더니 썩 나쁘지는 않은것 같네요 😢

Reply:
warning도 별도로 안 잡혀서 요렇게 사용해도 괜찮을 것 같네요 ㅎㅎ